### PR TITLE
#84 feat: add test in console

### PR DESCRIPTION
**Pull request by dawnzzz**
add test in console using rpc reflection

### CODE REVIEW

**Review by DMwangnima — COMMENTED** [review]
You can add testing files in both services and util directory and make use of TestMain to initialize a Reflection Server(dubbo-go or grpc), then test each method on by one.

This helps you verify the functionalities and improve the coverage.

**Comment by DMwangnima** [comment]
If you have any question about the initialization process or cases writing. Please let me know.

**Comment by dawnzzz** [comment]
基于gRPC reflection进行服务测试，基本流程如下：
1. 通过reflection获取信息，包括但不限于：
    - service name list
    - method name list
    - describe string（proto定义）
    - template string（输入/输出 json 模版）
2. 基于接口信息 invoke，支持unary，client stream，server stream，bi-stream模式，input format is json

应用侧需要：
- 对于gRPC应用，需要开启grpc reflection
- 对于dubbo应用，triple协议目前已经默认开启了reflection，并且与grpc reflection兼容，故可以通过grpc reflection获取接口信息（对于 dubbo go应用，测试于 [d2ed4f534bdfb14f59f795297c896cf18bb72cd4](https://github.com/apache/dubbo-go/tree/d2ed4f534bdfb14f59f795297c896cf18bb72cd4)，更早的版本可能会不支持 reflection/不兼容grpc reflection）

File: pkg/admin/util/reflection/reflection_test.go [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+)
+
+var (
+	tripleServer *server.Server
+	grpcServer   *grpc.Server
+	grpcRef      RPCReflection
+	tripleRef    RPCReflection
+)
+
+func init() {
```
Comment (DMwangnima) [inline pkg/admin/util/reflection/reflection_test.go:52]: You can use TestMain to initialize:
```
func TestMain(m *testing.M) {
    initGRpcServer()
    initTripleServer()
    initGrpcReflection()
    initTripleReflection()
    os.Exit(m.Run())
}
```

File: pkg/admin/util/reflection/reflection_test.go [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+
+	for _, list := range lists {
+		t.Run(list.name, func(t *testing.T) {
+			for _, method := range list.methods {
+				txt, err := list.ref.DescribeString(method)
+				if err != nil {
+					t.Errorf("error in DescribeString for %v", method)
+					return
+				}
+				t.Logf("name:%#v method:%#v txt:\n%#v", list.name, method, txt)
```
Comment (DMwangnima) [inline pkg/admin/util/reflection/reflection_test.go:234]: If you feel that verifying the results directly is tedious, you can determine if the results of grpc and triple are the same (assuming that grpc's results are correct).

File: pkg/admin/util/reflection/reflection_test.go [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+	for _, list := range lists {
+		t.Run(list.name, func(t *testing.T) {
+			for _, method := range list.methods {
+
+				inputType, outputType, err := list.ref.InputAndOutputType(method)
+				if err != nil {
+					t.Errorf("error in InputAndOutputType for %v", method)
+					return
+				}
+				t.Logf("name:%#v method:%#v inputType:%#v outputType:%#v", list.name, method, inputType, outputType)
```
Comment (DMwangnima) [inline pkg/admin/util/reflection/reflection_test.go:270]: ditto.

File: pkg/admin/util/reflection/reflection_test.go [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+					return
+				}
+				t.Logf("name:%#v inputType:%#v txt:\n%#v", list.name, inputType, txt)
+
+				txt, err = list.ref.DescribeString(outputType)
+				if err != nil {
+					t.Errorf("error in DescribeString for %v", outputType)
+					return
+				}
+				t.Logf("name:%#v outputType:%#v txt:\n%#v", list.name, outputType, txt)
```
Comment (DMwangnima) [inline pkg/admin/util/reflection/reflection_test.go:319]: ditto.

File: pkg/admin/util/reflection/reflection_test.go [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+					return
+				}
+				t.Logf("name:%#v inputType:%#v txt:\n%#v", list.name, inputType, txt)
+
+				txt, err = list.ref.TemplateString(outputType)
+				if err != nil {
+					t.Errorf("error in TemplateString for %v", outputType)
+					return
+				}
+				t.Logf("name:%#v outputType:%#v txt:\n%#v", list.name, outputType, txt)
```
Comment (DMwangnima) [inline pkg/admin/util/reflection/reflection_test.go:368]: ditto.

File: pkg/admin/util/reflection/reflection_test.go (hunk 2 above)
Comment (dawnzzz) [reply pkg/admin/util/reflection/reflection_test.go:234]: Get it! Thank you😊

**Comment by dawnzzz** [comment]
> LGTM. BTW, pls fix the lint problem.

I dont find lint problem, and I exec command 'make lint' in local which result is sussessful. 

I'm confused. I don't know what the problem is. 🤔️

**Comment by DMwangnima** [comment]
> > LGTM. BTW, pls fix the lint problem.
> 
> I dont find lint problem, and I exec command 'make lint' in local which result is sussessful.
> 
> I'm confused. I don't know what the problem is. 🤔️

cc @chickenlj .

**Comment by yuluo-yx** [comment]
IMO, you can running the command `/home/runner/golangci-lint-1.56.1-linux-amd64/golangci-lint run --out-format=github-actions --timeout=30m -v --disable-all --enable=gofumpt --enable=govet --enable=staticcheck --enable=ineffassign --enable=misspell` in local machine, maybe find some questions. @dawnzzz  🤔️

**Review by yuluo-yx — APPROVED** [review]
LGTM!

**Comment by dawnzzz** [comment]
> IMO, you can running the command `/home/runner/golangci-lint-1.56.1-linux-amd64/golangci-lint run --out-format=github-actions --timeout=30m -v --disable-all --enable=gofumpt --enable=govet --enable=staticcheck --enable=ineffassign --enable=misspell` in local machine, maybe find some questions. @dawnzzz 🤔️

Thank you for your help, but running the command locally without errors...

**Comment by chickenlj** [comment]
image

**Comment by chickenlj** [comment]
Above is where the lint error is reported.

**Pull request merged**